README.md

```diff
@@ -143,6 +143,79 @@ All protobuf message classes are generated and available. Key classes include:
 
 See the `examples/` directory for complete working examples.
 
+## Authentication
```

Comment:
These are good docs! Let's plan to move these to the docs repo. I say _move_ intentionally (_cut_ and paste) rather than copy, because I think we probably don't want to have to maintain them in two places. What do you think?